---
title: "Burger_Review"
author: "Issi Burger"
date: "2023-03-18"
output: html_document
---

Code runs perfectly! I really appreciate all of the notes throughout your .Rmd file. It really made it easy for me to understand what you were doing and gave me ideas on better ways to run my code in the future!

##Data Inspection
For data inspection, I really like the different commands you use to look at the data frames. Specifically, I like the "dim" command and how you used "sapply". I've been struggling on figuring out ways to use sapply and lapply, so it was nice to see it used in your code. Also, these two commands (see below) are great. Transforming all "multiple" and "unknown" values to NA is also a great idea as it makes removing them later super simple. 

```{r}
genotype %>% 
  group_by(Group) %>% 
  count()

pos %>% 
  group_by(Chromosome) %>% 
  summarise(Max=max(Position, na.rm = T), Min=min(Position, na.rm = T),    Number=length(Position))
```

##Data Processing
```{r}
maize <- genotype[which(genotype$Group=="ZMMIL" | genotype$Group =="ZMMLR" | genotype$Group == "ZMMMR"),]
teosinte <-genotype[which(genotype$Group=="ZMPBA" | genotype$Group =="ZMPIL" | genotype$Group == "ZMPJA"),]
```

-   Another method:

```{r, eval=FALSE}
maize <- genotype %>% filter(Group=="ZMMIL" | Group =="ZMMLR" | Group == "ZMMMR")
teosinte <-genotype %>% filter(Group=="ZMPBA" | Group =="ZMPIL" | Group == "ZMPJA")
```


I appreciate you providing two different methods on how to select these specific groups from the data sets. Another option is to subset the data using the following code: 

maize <- subset(fang_genotype, fang_genotype$Group==c("ZMMIL","ZMMLR","ZMMMR"))

Also, instead of specifying genotype$Group three times, you could potentially put them in a list (Group==c("ZMMIL","ZMMLR","ZMMMR")). I haven't tried the which or filter commands, so I'm not sure if a list would work, but just a thought!


```{r, eval = FALSE}
chr <- 1:10
for (i in chr) {
  files_inc <- maizewp[maizewp$Chromosome == i,]
  files_inc[files_inc == "?/?"] <- "?"
  if (i < 10) { write.table(files_inc, file = paste("Maize_Chr0",i,"_increase.txt",sep=""),row.names = FALSE,sep = "\t",quote = FALSE) }
  else {write.table(files_inc, file = paste("Maize_Chr",i,"_increase.txt",sep=""),row.names = FALSE, sep = "\t",quote = FALSE)}
  
  files_dec <- maizewp[maizewp$Chromosome == i,]
  files_dec[files_dec == "?/?"] <- "-"
  files_dec <- files_dec %>% arrange(desc(Chromosome),desc(Position))
  if (i < 10) { write.table(files_dec, file = paste("Maize_Chr0",i,"_decrease.txt",sep=""),row.names = FALSE,sep = "\t",quote = FALSE) }
  else {write.table(files_dec, file = paste("Maize_Chr",i,"_decrease.txt",sep=""),row.names = FALSE, sep = "\t",quote = FALSE)}
}
  

```

```{r, eval = FALSE}
chr <- 1:10
for (i in chr) {
  files_inc <- teosintewp[teosintewp$Chromosome == i,]
  files_inc[files_inc == "?/?"] <- "?"
  if (i < 10) { write.table(files_inc, file = paste("Teosinte_Chr0",i,"_increase.txt",sep=""),row.names = FALSE,sep = "\t",quote = FALSE) }
  else {write.table(files_inc, file = paste("Teosinte_Chr",i,"_increase.txt",sep=""),row.names = FALSE, sep = "\t",quote = FALSE)}
  
  files_dec <- teosintewp[teosintewp$Chromosome == i,]
  files_dec[files_dec == "?/?"] <- "-"
  files_dec <- files_dec %>% arrange(desc(Chromosome),desc(Position))
  if (i < 10) { write.table(files_dec, file = paste("Teosinte_Chr0",i,"_decrease.txt",sep=""),row.names = FALSE,sep = "\t",quote = FALSE) }
  else {write.table(files_dec, file = paste("Teosinte_Chr",i,"_decrease.txt",sep=""),row.names = FALSE, sep = "\t",quote = FALSE)}
}
```

I really like your use of for loops. This is a great way to separate the data by chromosome number and is a lot more efficient than writing the same code over and over again (which is what I did...). I especially like your use of the if-else command so your files are organized in their respective folders!

##Visualization

Your figures are soooo pretty, so excellent job on that! For the graph that shows SNP per chromosome, you probably don't need the legend since it's a repetition of your x-axis. For your graph showing homo/heterozygosity, I like that you separated it by group. My favorite of your figures, though, is the violin plot. It's a great figure choice to show the distribution of SNP positions. 


Overall, your code is great! I learned a lot just by looking at what you did, so thanks for that. Great job!
